auth/middleware.go

```diff
@@ -17,17 +18,26 @@ import (
 const DefaultPostgrestRole = "postgrest_api"
 
 type kratosMiddleware struct {
-	client     *client.APIClient
-	jwtSecret  string
-	tokenCache *cache.Cache
+	client             *client.APIClient
+	jwtSecret          string
+	tokenCache         *cache.Cache
+	authSessionCache   *cache.Cache
+	basicAuthSeparator string
 }
 
-func (k *KratosHandler) KratosMiddleware() *kratosMiddleware {
-	return &kratosMiddleware{
-		client:     k.client,
-		jwtSecret:  k.jwtSecret,
-		tokenCache: cache.New(3*24*time.Hour, 12*time.Hour),
+func (k *KratosHandler) KratosMiddleware() (*kratosMiddleware, error) {
+	randString, err := utils.GenerateRandString(30)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate random string: %w", err)
 	}
+
+	return &kratosMiddleware{
+		client:             k.client,
+		jwtSecret:          k.jwtSecret,
+		tokenCache:         cache.New(3*24*time.Hour, 12*time.Hour),
+		authSessionCache:   cache.New(30*time.Minute, time.Hour),
```

Comment:
@moshloop I could also potentially use the same cache instance and use custom expiration time for basic auth caches.